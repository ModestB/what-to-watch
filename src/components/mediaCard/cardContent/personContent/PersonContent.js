import React from "react";

import { connect } from "react-redux";

import useRoute from "../../../../hooks/useRoute";

import KnowForList from "../../knowForList/KnowForList";
import classes from "./PersonContent.module.scss";

export const PersonContent = (props) => {
  const { isSinglePage } = useRoute();
  let content = null;
  let name = null;
  let bodyContent = null;

  name = (
    <div
      className={`${[
        classes.name,
        !isSinglePage ? classes.nameShort : null,
      ].join(" ")}`}
    >
      {props.personName}
    </div>
  );

  if (props.profileKnownFor) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <KnowForList elements={props.profileKnownFor} />
      </div>
    );
  }

  if (isSinglePage) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.known_for_department
            ? props.profileDetails.known_for_department
            : "-"}
        </p>
        <p className={`${classes.title}`}>Birthday </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.birthday ? props.profileDetails.birthday : "-"}
        </p>
        <p className={`${classes.title}`}>Place Of Birth</p>
        <p className={`${classes.text}`}>
          {props.profileDetails.place_of_birth
            ? props.profileDetails.place_of_birth
            : "-"}
        </p>
      </div>
    );
  }

  content = (
    <React.Fragment>
      {name}
      {bodyContent}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    profileDetails: state.profileDetails,
  };
};

export default connect(mapStateProps, null)(PersonContent);
