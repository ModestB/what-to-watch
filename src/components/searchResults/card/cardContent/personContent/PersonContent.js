import React from "react";

import { connect } from "react-redux";

import KnowForList from "../../knowForList/KnowForList";
import classes from "./PersonContent.module.scss";

export const PersonContent = (props) => {
  let content = null;
  let name = null;
  let bodyContent = null;

  name = (
    <div
      className={`${[
        classes.name,
        !props.displaySinglePage ? classes.nameShort : null,
      ].join(" ")}`}
    >
      {props.personName}
    </div>
  );

  if (props.profileKnownFor) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <KnowForList elements={props.profileKnownFor} />
      </div>
    );
  }

  if (props.displaySinglePage) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.known_for_department
            ? props.profileDetails.known_for_department
            : "-"}
        </p>
        <p className={`${classes.title}`}>Birthday </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.birthday ? props.profileDetails.birthday : "-"}
        </p>
        <p className={`${classes.title}`}>Place Of Birth</p>
        <p className={`${classes.text}`}>
          {props.profileDetails.place_of_birth
            ? props.profileDetails.place_of_birth
            : "-"}
        </p>
      </div>
    );
  }

  content = (
    <React.Fragment>
      {name}
      {bodyContent}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    displaySinglePage: state.displaySinglePage,
    profileDetails: state.profileDetails,
  };
};

export default connect(
  mapStateProps,
  null
)(PersonContent);
