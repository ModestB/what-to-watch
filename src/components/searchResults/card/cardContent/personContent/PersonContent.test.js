import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { PersonContent } from "./PersonContent";

configure({ adapter: new Adapter() });

describe("<PersonContent />", () => {
  let wrapper;
  let props = {
    displaySinglePage: false,
    personName: "Sophie Turner",
    profileKnownFor: "Acting",
    profileDetails: {
      birthday: "1996-02-21",
      known_for_department: "Acting",
      deathday: null,
      id: 1001657,
      name: "Sophie Turner",
      also_known_as: ["소피 터너", "Σόφι Τέρνερ", "Софи Тёрнер"],
      gender: 1,
      biography:
        "Sophie Turner (born 21 February 1996) is an English actress. Turner made her professional acting debut as Sansa Stark on the HBO fantasy television series Game of Thrones (2011–2019), which brought her international recognition.\n\nTurner has also starred in the television film The Thirteenth Tale (2013) and she made her feature film debut in Another Me (2013). She has also starred in the action comedy Barely Lethal (2015) and portrays a young Jean Grey / Phoenix in the X-Men film series.",
      popularity: 8.512,
      place_of_birth: "Northampton, Northamptonshire, England",
      profile_path: "/5lzG6z74a8aYVWLsoAQVkyh5IEa.jpg",
      adult: false,
      imdb_id: "nm3849842",
      homepage: null
    }
  };

  beforeEach(() => {
    wrapper = shallow(<PersonContent {...props} />);
  });

  it("should match snapshot displaySinglePage = false", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it("should match snapshot displaySinglePage = true", () => {
    wrapper.setProps({ displaySinglePage: true });
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
