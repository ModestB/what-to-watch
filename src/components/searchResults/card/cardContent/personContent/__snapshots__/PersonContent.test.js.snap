// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonContent /> should match snapshot displaySinglePage = false 1`] = `
"<_default>
  <div className=\\"name nameShort\\">
    Sophie Turner
  </div>
  <div>
    <p className=\\"title\\">
      Known for 
    </p>
    <Connect(knowForList) elements=\\"Acting\\" />
  </div>
</_default>"
`;

exports[`<PersonContent /> should match snapshot displaySinglePage = true 1`] = `
"<_default>
  <div className=\\"name \\">
    Sophie Turner
  </div>
  <div>
    <p className=\\"title\\">
      Known for 
    </p>
    <p className=\\"text\\">
      Acting
    </p>
    <p className=\\"title\\">
      Birthday 
    </p>
    <p className=\\"text\\">
      1996-02-21
    </p>
    <p className=\\"title\\">
      Place Of Birth
    </p>
    <p className=\\"text\\">
      Northampton, Northamptonshire, England
    </p>
  </div>
</_default>"
`;
